ObservableProperty not generated for WPF application targeting .NET Framework 4.8

In the [Introduction to the MVVM Toolkit](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/) Microsoft Learn article (version from January 12, 2023 retrieved on May 12, 2023), it is stated that

> This package targets .NET Standard so it can be used on any app platform: UWP, WinForms, WPF, Xamarin, Uno, and more; and on any runtime: .NET Native, .NET Core, .NET Framework, or Mono. It runs on all of them.

which seems to indicate that this Community Toolkit should work in an application using WPF that targets .NET Framework 4.8.

If this is the case, then I believe there is a bug present in the version 8.2.0 NuGet package.

The issue is that the code for an `ObservableProperty` of an `ObservableObject` (such as in the example in the [ObservablePropertyAttribute Class documentation](https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.mvvm.componentmodel.observablepropertyattribute)) does not get generated when building a WPF application that targets .NET Framework 4.8.

The attached WpfApp1 Visual Studio 2022 Solution is a minimal reproducible example wherein `Class1` is an `ObservableObject` with an `ObservableProperty` and the `App.OnStartup` method attempts to use the `Class1.Name` property that is supposed to be generated.

Attempting to Build Solution yields

> error CS1061: 'Class1' does not contain a definition for 'Name' and no accessible extension method 'Name' accepting a first argument of type 'Class1' could be found (are you missing a using directive or an assembly reference?)

and is preventing my utilization of this feature.

[WpfApp1.zip](https://github.com/CommunityToolkit/dotnet/files/11467953/WpfApp1.zip)

--- a/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
+++ b/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
@@ -102,15 +102,19 @@
     </ItemGroup>
   </Target>
 
-  <!-- Remove the analyzer if packages.config is used -->
+  <!--
+    Inform the user if packages.config is used (as the analyzers and the source generators
+    won't work at all). Since packages.config can only be used with legacy-style projects,
+    the entire package can be skipped if an SDK-style project is used.
+  -->
   <Target Name="MVVMToolkitWarnForPackagesConfigUse"
           AfterTargets="ResolvePackageDependenciesForBuild;ResolveNuGetPackageAssets"
-          DependsOnTargets="MVVMToolkitGatherAnalyzers">
+          Condition="'$(UsingMicrosoftNetSDK)' != 'true'">
     
     <!--
       Check whether packages are being restored via packages.config, by reading the associated MSBuild property.
       This happens when either the project style is using packages.config, or when explicitly requested.
-      See https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-projects-with-msbuild.
+      See https://learn.microsoft.com/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-projects-with-msbuild.
     -->
     <PropertyGroup>
       <MVVMToolkitIsTargetProjectUsingPackagesConfig Condition ="'$(RestorePackagesConfig)' == 'true' OR '$(RestoreProjectStyle)' == 'PackagesConfig'">true</MVVMToolkitIsTargetProjectUsingPackagesConfig>
@@ -120,6 +124,8 @@
       If no packages.config properties are set, also try to manually find the packages.config file.
       This will be in the @(None) elements, if present. Doing so makes sure this works in builds as
       well, since the implicit targets populating the properties above only run when restoring.
+      Since the packages.config file will always be in the root of the project, if present, we will
+      match with the full item spec (see https://learn.microsoft.com/nuget/reference/packages-config).
     -->
     <FindInList ItemSpecToFind="packages.config"
                 List="@(None)"

--- a/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
+++ b/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
@@ -54,7 +54,10 @@
       emitting this manually lets us customize the message to inform developers as to why exactly the generators have been
       disabled, and that the rest of the MVVM Toolkit will still keep working as intended, just without additional features.
     -->
-    <Warning Condition ="'$(MVVMToolkitCurrentCompilerVersionIsNotNewEnough)' == 'true'" Text="The MVVM Toolkit source generators have been disabled on the current configuration, as they need Roslyn 4.x in order to work. The MVVM Toolkit will work just fine, but features relying on the source generators will not be available."/>
+    <Warning Condition ="'$(MVVMToolkitCurrentCompilerVersionIsNotNewEnough)' == 'true'"
+             Code="MVVMTKCFG0001"
+             HelpLink="https://aka.ms/mvvmtoolkit/errors/mvvmtkcfg0001"
+             Text="The MVVM Toolkit source generators have been disabled on the current configuration, as they need Roslyn 4.x in order to work. The MVVM Toolkit will work just fine, but features relying on the source generators will not be available."/>
   
     <PropertyGroup>
 
@@ -109,7 +112,10 @@
       This happens when either the project style is using packages.config, or when explicitly requested.
       See https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-projects-with-msbuild.
     -->
-    <Warning Condition ="'$(RestorePackagesConfig)' == 'true' OR '$(RestoreProjectStyle)' == 'PackagesConfig'" Text="The MVVM Toolkit source generators might not be loaded correctly, as the current project is using the packages.config setup to restore NuGet packages. Source generators require PackageReference to be used (either in a legacy-style or SDK-style .csproj project, both are supported as long as PackageReference is used)."/>
+    <Warning Condition ="'$(RestorePackagesConfig)' == 'true' OR '$(RestoreProjectStyle)' == 'PackagesConfig'"
+             Code="MVVMTKCFG0002"
+             HelpLink="https://aka.ms/mvvmtoolkit/errors/mvvmtkcfg0002"
+             Text="The MVVM Toolkit source generators might not be loaded correctly, as the current project is using the packages.config setup to restore NuGet packages. Source generators require PackageReference to be used (either in a legacy-style or SDK-style .csproj project, both are supported as long as PackageReference is used)."/>
   </Target>
 
 </Project>

--- a/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
+++ b/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
@@ -99,4 +99,17 @@
     </ItemGroup>
   </Target>
 
+  <!-- Remove the analyzer if packages.config is used -->
+  <Target Name="MVVMToolkitWarnForPackagesConfigUse"
+          AfterTargets="ResolvePackageDependenciesForBuild;ResolveNuGetPackageAssets"
+          DependsOnTargets="MVVMToolkitGatherAnalyzers">
+
+    <!--
+      Emit a warning in case packages.config is used, by reading the associated MSBuild property.
+      This happens when either the project style is using packages.config, or when explicitly requested.
+      See https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-projects-with-msbuild.
+    -->
+    <Warning Condition ="'$(RestorePackagesConfig)' == 'true' OR '$(RestoreProjectStyle)' == 'PackagesConfig'" Text="The MVVM Toolkit source generators might not be loaded correctly, as the current project is using the packages.config setup to restore NuGet packages. Source generators require PackageReference to be used (either in a legacy-style or SDK-style .csproj project, both are supported as long as PackageReference is used)."/>
+  </Target>
+
 </Project>

--- a/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
+++ b/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
@@ -106,13 +106,35 @@
   <Target Name="MVVMToolkitWarnForPackagesConfigUse"
           AfterTargets="ResolvePackageDependenciesForBuild;ResolveNuGetPackageAssets"
           DependsOnTargets="MVVMToolkitGatherAnalyzers">
-
+    
     <!--
-      Emit a warning in case packages.config is used, by reading the associated MSBuild property.
+      Check whether packages are being restored via packages.config, by reading the associated MSBuild property.
       This happens when either the project style is using packages.config, or when explicitly requested.
       See https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-projects-with-msbuild.
     -->
-    <Warning Condition ="'$(RestorePackagesConfig)' == 'true' OR '$(RestoreProjectStyle)' == 'PackagesConfig'"
+    <PropertyGroup>
+      <MVVMToolkitIsTargetProjectUsingPackagesConfig Condition ="'$(RestorePackagesConfig)' == 'true' OR '$(RestoreProjectStyle)' == 'PackagesConfig'">true</MVVMToolkitIsTargetProjectUsingPackagesConfig>
+    </PropertyGroup>
+
+    <!--
+      If no packages.config properties are set, also try to manually find the packages.config file.
+      This will be in the @(None) elements, if present. Doing so makes sure this works in builds as
+      well, since the implicit targets populating the properties above only run when restoring.
+    -->
+    <FindInList ItemSpecToFind="packages.config"
+                List="@(None)"
+                MatchFileNameOnly="false"
+                Condition="'$(MVVMToolkitIsTargetProjectUsingPackagesConfig)' != 'true'">
+      <Output TaskParameter="ItemFound" PropertyName="MVVMToolkitPackagesConfigFile"/>
+    </FindInList>
+
+    <!-- Make sure to update the MSBuild property if the above task did find something -->
+    <PropertyGroup>
+      <MVVMToolkitIsTargetProjectUsingPackagesConfig Condition ="'$(MVVMToolkitPackagesConfigFile)' == 'packages.config'">true</MVVMToolkitIsTargetProjectUsingPackagesConfig>
+    </PropertyGroup>
+
+    <!-- Emit a warning in case packages.config is used -->
+    <Warning Condition ="'$(MVVMToolkitIsTargetProjectUsingPackagesConfig)' == 'true'"
              Code="MVVMTKCFG0002"
              HelpLink="https://aka.ms/mvvmtoolkit/errors/mvvmtkcfg0002"
              Text="The MVVM Toolkit source generators might not be loaded correctly, as the current project is using the packages.config setup to restore NuGet packages. Source generators require PackageReference to be used (either in a legacy-style or SDK-style .csproj project, both are supported as long as PackageReference is used)."/>

